tests/components/generic/test_camera.py

```diff
@@ -261,31 +274,38 @@ async def test_stream_source_error(hass, hass_client, hass_ws_client, fakeimgbyt
         }
 
 
-async def test_setup_alternative_options(hass, hass_ws_client):
+@respx.mock
+async def test_setup_alternative_options(
+    hass, hass_ws_client, fakeimgbytes_png, mock_av_open
+):
     """Test that the stream source is setup with different config options."""
-    assert await async_setup_component(
-        hass,
-        "camera",
-        {
-            "camera": {
-                "name": "config_test",
-                "platform": "generic",
-                "still_image_url": "https://example.com",
-                "authentication": "digest",
-                "username": "user",
-                "password": "pass",
-                "stream_source": "rtsp://example.com:554/rtsp/",
-                "rtsp_transport": "udp",
+    respx.get("https://example.com").respond(stream=fakeimgbytes_png)
+
+    with mock_av_open:
+        assert await async_setup_component(
+            hass,
+            "camera",
+            {
+                "camera": {
+                    "name": "config_test",
+                    "platform": "generic",
+                    "still_image_url": "https://example.com",
+                    "authentication": "digest",
+                    "username": "user",
+                    "password": "pass",
+                    "stream_source": "rtsp://example.com:554/rtsp/",
+                    "rtsp_transport": "udp",
+                },
             },
-        },
-    )
-    await hass.async_block_till_done()
+        )
+        await hass.async_block_till_done()
     assert hass.data["camera"].get_entity("camera.config_test")
```

Comment:
Please don't access hass.data in tests.

https://developers.home-assistant.io/docs/development_testing?_highlight=testing#writing-tests-for-integrations

Please use `hass.states.get` instead

Reply:
Sure, but that is existing code was already there.  Will put into a separate PR.

Reply:
#68316